go/vt/automation/resharding_task.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package automation
+
+import (
+	"fmt"
+	"strings"
+
+	pb "github.com/youtube/vitess/go/vt/proto/automation"
+	"github.com/youtube/vitess/go/vt/topo"
+)
+
+// ReshardingTask is a cluster operation which allows to increase the number of shards.
+type ReshardingTask struct {
+}
+
+func splitShardListIntoKeyspaceAndShards(keyspaceAndShards []string) (string, []string, error) {
+	var firstKeyspace string
+	shards := make([]string, 0, len(keyspaceAndShards))
+
+	for i, keyspaceAndShard := range keyspaceAndShards {
+		keyspace, shard, err := topo.ParseKeyspaceShardString(keyspaceAndShard)
+		if err != nil {
+			return "", nil, err
+		}
+
+		if i == 0 {
+			firstKeyspace = keyspace
+		} else {
+			if firstKeyspace != keyspace {
+				return "", nil, fmt.Errorf("All shards must have the same keyspace. First seen keyspace: %v Wrong shard: %v", firstKeyspace, keyspaceAndShard)
+			}
+		}
+		shards = append(shards, shard)
+	}
+
+	return firstKeyspace, shards, nil
+}
+
+func selectAnyTabletFromShardByType(shard string, tabletType string) string {
+	return ""
+}
+
+func (t *ReshardingTask) run(parameters map[string]string) ([]*pb.TaskContainer, string, error) {
+	sourceShardsWithKeyspace := strings.Split(parameters["source_shard_list"], ",")
```

Comment:
These parameters are a bit confusing to me, I'd rather have (with examples):
keyspace = ruser
source_shard_list = 10-20
dest_shard_list = 10-18,18-20